### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by newhoteng**
## OOP school library: decorate a class
In this milestone, I:
- Created a class `Nameable`
  - Implemented `correct_name` method that raises a `NotImplementedError`
- `Person` class is made to inherit from the `Nameable` class
  - `correct_name` method that returns the @name attribute implemented
- Created a `Decorator` class that inherits from the `Nameable` class
  - A nameable object passed as a parameter to the constructor method is assigned to an instance variable @nameable
  - `correct_name` method that returns the results of the correct_name method of the @nameable is implemented
- Created a `CapitalizeDecorator` class that in inherits from the `Decorator` class
   - `correct_name` method that capitalizes the output of `@nameable.correct_name` is implemented
- Created a `TrimmerDecorator` class that in inherits from the `Decorator` class
   - `correct_name` method that makes sure that the output of @nameable.correct_name has a maximum of 10 characters is implemented
- `testcode.rb` is added with sample code to test created classes and methods

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (omarsalem7) [inline decorator.rb:6]: - [ ] Kindly consider adding `attr_accessor :nameable` to allow access to instance variables from outside the scope of the class definition

File: trimmer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
```
Comment (omarsalem7) [inline trimmer.rb]: - [ ] As stated in the project requirement in the following screen, kindly consider the name has a maximum of 10 characters. If it's longer it should trim the word. 
```ruby
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```
>image

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @newhoteng 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Create the `Nameable` class properly ✔️ 
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @newhoteng, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**